docs/source/building/shared-libraries.rst

```diff
@@ -0,0 +1,23 @@
+Using shared libraries
+======================
+
+**Shared libraries** are libraries that are loosely coupled to the programs and extensions that depend on them. When loading an executable into memory, an operating system will find all dependent shared libraries and link them to the executable so that it can run. The major operating systems Linux, OS X, and Windows all provide a way to build executables and libraries that contain links to the shared libraries they depend on, instead of the directly linking the libraries themselves.
+
+Dynamic links are discovered in a similar manner to the way that python modules are discovered via ``PYTHONPATH``, and executables are discovered via ``PATH``.  A list of search locations is made, then the library objects are searched for in the search locations.  By default, as well as by design, the system dynamic linker does not have any special preference for the conda environment lib directories.
+
+Both absolute links and relative links may be specified. If the links were absolute paths like ``/Users/jsmith/my_build_env`` then the library would only work on a system where that exact path exists. Therefore, relative links are preferred in conda packages.
+
+To use relative links, we need to use a special variable in the link itself.  In Linux, the $ORIGIN variable allows you to say "relative to this file as it's being executed".  On OS X, there are ``@rpath``, ``@loader_path``, ``@executable_path``.  ``@rpath`` allows you to set relative links from the system load paths, while ``@loader_path`` is equivalent to ``$ORIGIN``, and ``@executable_path`` supports the Apple ".app" directory approach, where libraries know where they live relative to their calling application.  In Windows, _____________.
```

Comment:
Windows does not actually have this concept of embedding links into binaries, and it stinks.  Here's the rules for Windows:

https://msdn.microsoft.com/en-us/library/7d83bc18.aspx

We dump most (hopefully eventually all) DLLs into (install prefix)\Library\bin.  This now gets added to os.environ["PATH"] for all Python processes:
https://github.com/ContinuumIO/anaconda/commit/2b61a39943f30f1609b651d6f913043bd86047c6
https://github.com/ContinuumIO/anaconda/commit/42aa701386da6fb7d5df4e0e05bb8d999e4ea230

The lack of this before cause major issues with the rollout of Anaconda 2.4: installers add that to the system (or user) PATH, but upgrades did not run installers.  Thus Library/bin was not on PATH, and the most common symptom was that matplotlib failed to load libpng (DLL import failed).